Source/ElasticLINQ/ElasticQueryProvider.cs

```diff
@@ -123,7 +123,22 @@ public object Execute(Expression expression)
                         throw new InvalidOperationException("No HTTP response received.");
                 }
 
-                return translation.Materializer.Materialize(response);
+                var result = translation.Materializer.Materialize(response);
+
+                if (response.hits != null)
+                {
+                    var hits = response.hits.hits;
+                    if (hits != null && hits.Capacity > 4096)
```

Comment:
Probably not. I guess zeroing out an array should be fairly well optimized in .net.